pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/PartitionSegmentPruner.java

```diff
@@ -151,10 +170,18 @@ public synchronized void refreshSegment(String segment) {
     } else {
       _partitionInfoMap.remove(segment);
     }
+    rebuildReverseMap();
   }
 
+//  /**
+//   * Package-level function to make previous tests happy
+//   */
+//  @VisibleForTesting
+//  SelectedSegments prune(BrokerRequest brokerRequest, Set<String> segments) {
```

Comment:
@Jackie-Jiang because of the signature change below, the Test has lots of changes.
If we add a function here like the one in the comment, we can avoid lots of changes in Test but this method will be only for testing simplicity.
I will remove this comment if we think the tests code change should be like what we have now